### PR TITLE
Exception-based context

### PR DESCRIPTION
**Pull request by nateberkopec**
Re: #479

### CODE REVIEW

File: lib/raven/event.rb [hunk 1]
```diff
@@ -69,6 +69,119 @@ def initialize(init = {})
...
+            evt.interface(:stacktrace) do |int|
+              stacktrace_interface_from(int, evt, options[:backtrace])
+            end
+          end
+        end
+      end
+
+      private
+
+      def get_exception_context(exc)
```
Comment (nateberkopec) [inline lib/raven/event.rb]: hard to see in this diff, but here's the new important method

**Comment by nateberkopec** [comment]
Gonna add docs, fix the build, then I think this is good to go.

**Comment by juanibiapina** [comment]
We want to use annotate_exception and it seems this patch fixes a bug we've found (the need for the DeepMerge there). Without the deep merge, extra data we pass to annotate exception will be overwritten by the options passed to Event.from_exception.

Any plans on getting this merged?

**Comment by nateberkopec** [comment]
Yes - it'll be in this week, I promise!

**Comment by juanibiapina** [comment]
@nateberkopec hey, sorry to push you, but I took liberty to look at the tests and it seems the needed changes are merely style. Would it be possible to release it any time soon?

**Comment by nateberkopec** [comment]
@juanibiapina yup, doing more work today and then its probably new version time

**Comment by juanibiapina** [comment]
@nateberkopec Would you release a version with these changes if your current work is not yet finished?

**Pull request merged**